### PR TITLE
has slug: allow multiple slug'd attributes per record

### PR DESCRIPTION
**Pull request by kellyredding**
The goal here is to have multiple attributes on a given record that
use the has slug logic to drive their values.  Very little is changed
from the original API - you still call `has_slug` as before and the
attribute name is defaulted as before if no custom attr name is given.

The biggest changes are the following:

* you now have the possibility of multiple has-slug configs per record,
  therefore the class method has been pluralized to `ardb_has_slug_configs`
  and is a hash of configs keyed by their attr name
* the `:attribute` key is no longer stored in the has-slug config -
  it is now given everywhere you need to lookup a config or the default
  is assumed
* calling `reset_slug` now assumes the default attr - optionally call
  it with the appropriate attr name to reset the non-default attrs

One implementation note: I didn't change the tests except for the
changes outlined above.  I got everything passing by changing how
the config was referenced in the tests and changing to pluralized
method names where needed.  Once everything passed with those
superficial changes, I then added additional tests to verify calling
`reset_slug` with and without a given attr name to verify this new
behavior case.

Update: As follow up to code review, I updated the tests further to call
`has_slug` multiple times, both with a custom attr like before and
with the default attr.  I then updated the tests to show that updating
the slugs updates all has-slug attrs configured (per feedback from
Collin).  As part of this change, I decided to take out the default
values on the internal `Slug` util class.  It is inconsistent that the
previous defaults used the higher order has-slug default separator but
used a custom default preprocessor.  This inconsistency showed when I
updated the tests and I had to do custom values to get a default exp
slug.  I decided to just remove the magic and make you pass in the 
separator and preprocessor for all calls (including the tests).  In
the live code we always pass the separator and preprocessor anyway.

@jcredding ready for review.  Do you see any issues with the class method name change (this is the only 'public' API change here I think)?

### CODE REVIEW

**Comment by kellyredding** [comment]
@Criptak - FYI.

File: lib/ardb/has_slug.rb [hunk 1]
```diff
@@ -53,26 +52,28 @@ module InstanceMethods
...
       end
 
-      def ardb_has_slug_generate_slug
-        attr_name = self.class.ardb_has_slug_config[:attribute]
-        slug_source = if !self.send(attr_name) || self.send(attr_name).to_s.empty?
-          self.instance_eval(&self.class.ardb_has_slug_config[:source_proc])
-        else
-          self.send(attr_name)
+      def ardb_has_slug_generate_slugs
+        self.class.ardb_has_slug_configs.keys.each do |attr_name|
```
Comment (jcredding) [inline lib/ardb/has_slug.rb]: @kellyredding - You might do: `self.class.ardb_has_slug_configs.each do |attr_name, config|`. Then you don't have to lookup the config.

**Comment by jcredding** [comment]
@kellyredding - As you kinda mentioned, you might add a test that shows when you instance eval `ardb_has_slug_generate_slugs` that it will generate a slug for each configuration. That would be the only test I might add for showing that it supports multiple configs.

**Comment by jcredding** [comment]
@kellyredding - A couple comments but overall I'm good with this :boom:

**Comment by kellyredding** [comment]
@jcredding update with your feedback - thanks.  Note the newly added paragraph of explanation on the commit msg and larger set of changes to the tests.  You cool with all this?

**Comment by jcredding** [comment]
@kellyredding - I'm good with the changes :boom:

**Pull request merged**